lib/bokken/ecto/pt_mobile.ex

```diff
@@ -13,66 +13,46 @@ defmodule Bokken.Ecto.PtMobile do
   ## Parameters
     - number: valid PT phone number with the "+351" indicative in a string format
   """
-  def cast("+351" <> number) do
-    mobile =
-      format(number)
-      |> List.to_string()
-
-    if is_nil(Regex.run(~r/9[12356]\d{7}/, mobile)) do
-      {:error, [message: "número PT não válido"]}
-    else
-      {:ok, "+351" <> mobile}
+  def cast(number) do
+    case format(number) do
+      {:ok, number} -> {:ok, number}
+      {:error, message} -> {:error, message}
     end
   end
 
-  def cast(_number) do
-    {:error, [message: "número PT não válido"]}
-  end
-
   @doc """
   Transforms the data into a specific format to be stored
 
   ## Parameters
     - number: valid PT phone number with the "+351" indicative in a string format
   """
-  def dump("+351" <> mobile) do
-    {:ok, "+351" <> mobile}
+  def dump(number) do
+    case format(number) do
+      {:ok, number} -> {:ok, number}
+      {:error, _message} -> :error
+    end
   end
 
-  def dump(_mobile), do: :error
-
   @doc """
   Transforms the data back to a runtime format
 
   ## Parameters
     - number: valid PT phone number with the "+351" indicative in a string format
   """
-  def load("+351" <> mobile) do
-    {:ok, "+351" <> mobile}
+  def load(number) do
+    case format(number) do
+      {:ok, number} -> {:ok, number}
+      {:error, _message} -> :error
+    end
   end
 
-  def load(_mobile), do: :error
-
-  # Transforms a number into the standard format
-  # ## Parameters
-  #   - number: valis PT phone number in a list of strings (ex: ["9", "1", "6", "0", "6", "5", "5", "0", "0"])
-  defp format([]), do: []
-
-  defp format(number) when is_list(number) do
-    [n | rest] = number
-    <<v::utf8>> = n
+  defp format(number) do
+    {:ok, phone_number} = ExPhoneNumber.parse(number, "PT")
 
-    if v >= 48 and v <= 57 do
-      [n | format(rest)]
+    if ExPhoneNumber.is_valid_number?(phone_number) do
+      {:ok, ExPhoneNumber.format(phone_number, :e164)}
     else
-      format(rest)
+      {:error, [message: "invalid PT phone number"]}
```

Comment:
```suggestion
      {:error, [message: gettext("invalid PT phone number")]}
```

So we keep the possibility to have translations :pray: